src/Importify/Resolution.hs

```diff
@@ -20,16 +23,23 @@ import           Importify.Common                   (Identifier (..),
                                                      importSpecToIdentifiers)
 
 symbolByName :: String -> [N.Symbol] -> Maybe N.Symbol
-symbolByName name symbols = head $ do
-    symbol <- symbols
-    guard $ symbolName symbol == stringToName name
-    pure symbol
+symbolByName name =
+    listToMaybe . filter ((stringToName name ==) . symbolName)
 
-symbolUsages :: N.Symbol -> [Scoped l] -> Maybe l
-symbolUsages symbol annotations = head $ do
-    Scoped (GlobalSymbol globalSymbol _) location <- annotations
-    guard (globalSymbol == symbol)
-    pure location
+symbolUsed :: N.Symbol -> [Scoped l] -> Bool
+symbolUsed symbol annotations = any used annotations
+  where
+    used :: Scoped l -> Bool
```

Comment:
I would like to see some comments near this `used` function. It's not obvious, why all these checks are required...